indexer/src/executor.rs

```diff
@@ -34,6 +43,96 @@ pub struct IndexEnv {
     pub db: Arc<Mutex<Database>>,
 }
 
+pub struct CustomIndexExecutor {
+    events: HashMap<ReceiptEvent, Vec<CustomHandler>>,
+    db: Arc<Mutex<Database>>,
+    #[allow(dead_code)]
+    manifest: Manifest,
+}
+
+impl CustomIndexExecutor {
+    pub fn new(db_conn: &str, manifest: Manifest) -> IndexerResult<Self> {
+        let events = HashMap::new();
+        let db = Arc::new(Mutex::new(Database::new(db_conn).unwrap()));
+
+        db.lock()
+            .expect("Lock poisoned from schema load")
```

Comment:
Didn't improve the error messaging. Figured we could do this in a follow-up pass, in a smaller PR (as this one is rather large, as is)